include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -696,6 +696,18 @@ typedef void (*OrtCustomJoinThreadFn)(OrtCustomThreadHandle ort_custom_thread_ha
 
 typedef OrtStatus*(ORT_API_CALL* RegisterCustomOpsFn)(OrtSessionOptions* options, const OrtApiBase* api);
 
+/** \brief Callback function for RunAsync
+ *
+ * \param[in] user_data A customized handle passed in by RunAsync.
+ * \param[out] outputs On succeed, outputs host inference results.
+               NOTE:
+               1. Ort is in charge of the lifetime of "outputs" array, but NOT each of its element, which is a OrtValue*.
+               2. Customer is expected to release each element of "outputs", which is a OrtValue*.
+ * \param[out] num_outputs Number of output.
+ * \param[out] status On error, status will provide details.
+ */
+typedef void (*RunAsyncCallbackFn)(void* user_data, OrtValue** outputs, size_t num_outputs, OrtStatusPtr status);
```

Comment:
Can also point out that this makes sense only in the case where AzureEP is involved, otherwise there's no need to use async.

Reply:
Why?
We made it very explicit:
https://github.com/microsoft/onnxruntime/blob/6c67e44890e70ace95f829829ce0adae0ef4d408/include/onnxruntime/core/session/onnxruntime_c_api.h#L4329
So it applies to desktops, and mobiles over many EPs?